src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureKeyVaultReference/AzureKeyVaultSecretProvider.cs

```diff
@@ -50,7 +51,23 @@ public async Task<string> GetSecretValue(Uri secretUri, string key, Cancellation
             else if (client != null)
             {
                 KeyVaultSecret secret;
-                secret = await client.GetSecretAsync(secretName, secretVersion, cancellationToken).ConfigureAwait(false);
+                bool success = false;
+
+                try
+                {
+                    secret = await client.GetSecretAsync(secretName, secretVersion, cancellationToken).ConfigureAwait(false);
+                    success = true;
+                }
+                finally
+                {
+                    // If this is not already a cached secret, we don't have any refresh time to update,
+                    // i.e. there's no way to track next refresh time of new secrets that fail to resolve.
```

Comment:
When this actually happens, shouldn't we cache a null secret and extend the expiration anyway (as if it was a successful case)? I know this shouldn't happen, but this makes sure we don't have a code path that doesn't extend the expiration.

Reply:
`null` is a valid secret value. If we start caching `null` for failed secrets, we won't be able to track which secret failed to load previously.

Reply:
I don't think we need to differentiate. If the previous attempt failed, we would not try again until the "backoff" time passes again. Isn't that what you want anyway?

Reply:
That makes sense because as far as the application is concerned, null secret value means that the secret is unusable. So we can say that the secret value can be `null` if:
- The secret was not found in KeyVault.
- The secret was found but its value is `null`.
- The secret failed to load due to an exception.
 
Also, right now if we fail to load a secret from KeyVault due to an exception, we would just never try to load the secret again - until its time to refresh another cached secret, or until RefreshAll is triggered. But if we start caching failed secrets as `null` and tracking their next refresh time, this would no longer be an issue.
Thanks for the great suggestion!

Reply:
Updated